### PR TITLE
Issue SB-28285 feat:Project template components design

### PR DESCRIPTION
**Pull request by vishwanath1004**

### CODE REVIEW

File: src/app/manage-learn/core/services/utils.service.ts [hunk 1]
```diff
@@ -219,6 +221,7 @@ export class UtilsService {
     } else {
       status = statusType.notStarted;
     }
+    this.numberTasksCompleted = completedList;
```
Comment (kiranharidas187) [inline src/app/manage-learn/core/services/utils.service.ts:224]: create a separate function for percentage calculation

File: src/app/manage-learn/core/services/utils.service.ts [hunk 2]
```diff
@@ -716,4 +719,40 @@ async getSortTasks(project:any) {
   return data;
 }
 
+
+getMetaDataActions(){
```
Comment (kiranharidas187) [inline src/app/manage-learn/core/services/utils.service.ts:723]: put this in to action constant file

File: src/app/manage-learn/project/project-templateview/project-templateview.page.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<ion-content  class="ion-padding">
+<app-metadata-details translucent="true" [data]="metaData">
+  <div class="sb-dt-card-actions">
+    <div class="wrapper">
+      <div>
+        <button *ngFor="let actionBtn of actionItems" (click)="action(actionBtn.action)" slot="icon-only" class="card-button">
```
Comment (kiranharidas187) [inline src/app/manage-learn/project/project-templateview/project-templateview.page.html:6]: create a separate component

File: src/app/manage-learn/shared/components/metadata-details/metadata-details.component.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+@import 'src/assets/styles/_custom-mixins';
```
Comment (kiranharidas187) [inline src/app/manage-learn/shared/components/metadata-details/metadata-details.component.scss:1]: if not require rm

File: src/app/manage-learn/shared/components/metadata-details/metadata-details.component.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit,Input } from '@angular/core';
+
+@Component({
+  selector: 'app-metadata-details',
+  templateUrl: './metadata-details.component.html',
+  styleUrls: ['./metadata-details.component.scss'],
+})
+export class MetadataDetailsComponent implements OnInit {
+  taskLength  = 3;
```
Comment (kiranharidas187) [inline src/app/manage-learn/shared/components/metadata-details/metadata-details.component.ts:9]: rm

File: src/app/manage-learn/shared/shared.module.ts [hunk 6]
```diff
@@ -67,6 +68,7 @@ import { ExpansionPanelComponent } from './components/expansion-panel/expansion-
 import { ExpansionTableComponent } from './components/expansion-table/expansion-table.component';
 import { GenericPopUpService } from './generic.popup';
 import { HTTP } from '@ionic-native/http/ngx';
+import { ComponentsModule } from '@app/app/components/components.module';
```
Comment (kiranharidas187) [inline src/app/manage-learn/shared/shared.module.ts:71]: rm

File: src/app/manage-learn/core/constants/actions.constants.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
+export const actions ={
+    PROJECTACTIONS:[
```
Comment (kiranharidas187) [inline src/app/manage-learn/core/constants/actions.constants.ts:2]: PROJECT_ACTIONS

File: src/app/manage-learn/project/project-templateview/project-templateview.page.html [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<ion-content  class="ion-padding">
+<app-metadata-details translucent="true" [data]="metaData">
+  <div class="sb-dt-card-actions">
+    <div class="wrapper"> 
+      <app-metadata-actions *ngFor="let actionBtn of actionItems" (actionEvent)="action($event)" [actionItem]="actionBtn"></app-metadata-actions>
```
Comment (kiranharidas187) [inline src/app/manage-learn/project/project-templateview/project-templateview.page.html:5]: can we handle this loop also inside the actions component?

Comment (vishwanath1004) [reply src/app/manage-learn/project/project-templateview/project-templateview.page.html:5]: If we add loop inside that component then, it is difficult to handle download and downloaded of project action and we are using same component for tasks also.

**Pull request merged**